dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
-                settings = $.extend({}, $.multiselect.defaults, $this.data(), (typeof options === 'object' && options));
-
-            if (!data) {
-                $this.data('crlcu.multiselect', (data = new Multiselect($this, settings)));
-            }
-        });
-    };
-
-    // append options
-    // then set the selected attribute to false
-    $.fn.move = function( $options ) {
-        this
-            .append($options)
-            .find('option')
-            .prop('selected', false);
-
-        return this;
-    };
-
-    $.fn.removeIfEmpty = function() {
-        if (!this.children().length) {
-            this.remove();
-        }
-
-        return this;
-    };
-
-    $.fn.mShow = function() {
-        this.removeClass('hidden').show();
-
-        if (isIE || isSafari) {
-            this.each(function(index, option) {
-                // Remove <span> to make it compatible with IE
-                if($(option).parent().is('span')) {
-                    $(option).parent().replaceWith(option);
-                }
-
-                $(option).show();
-            });
-        }
-
-        return this;
-    };
-
-    $.fn.mHide = function() {
-        this.addClass('hidden').hide();
-
-        if (isIE || isSafari) {
-            this.each(function(index, option) {
-                // Wrap with <span> to make it compatible with IE
-                if(!$(option).parent().is('span')) {
-                    $(option).wrap('<span>').hide();
-                }
-            });
-        }
-
-        return this;
-    };
-
-    // sort options then reappend them to the select
-    $.fn.mSort = function(callback) {
-        this
-            .children()
-            .sort(callback)
-            .appendTo(this);
-
-        this
-            .find('optgroup')
-            .each(function(i, group) {
-                $(group).children()
-                    .sort(callback)
-                    .appendTo(group);
-            })
-
-        return this;
-    };
-
-    // attach index to children
-    $.fn.attachIndex = function() {
-        this.children().each(function(index, option) {
-            var $option = $(option);
-
-            if ($option.is('optgroup')) {
-                $option.children().each(function(i, children) {
-                    $(children).data('position', i);
-                });
-            }
-
-            $option.data('position', index);
-        });
-    };
-
-    $.expr[":"].search = function(elem, index, meta) {
-        var regex = new RegExp(meta[3], "i");
-
-        return $(elem).text().match(regex);
-    }
-}));
-
+/*
+ * @license
+ *
+ * Multiselect v2.3.12
+ * http://crlcu.github.io/multiselect/
+ *
+ * Copyright (c) 2016 Adrian Crisan
+ * Licensed under the MIT license (https://github.com/crlcu/multiselect/blob/master/LICENSE)
+ */
+
+/**
+ * The jQuery object.
+ * @external jQuery
+ * */
+
+/**
+ * The jQuery prototype alias.
+ * @external jQuery.fn
+ */
+
+/**
```

Comment:
I define several objects here so that I can document their properties and default values. And I can later reference them easier.